### PR TITLE
add test for openfold triton mha kernel

### PR DESCRIPTION
**Pull request by azazhu**
add test for openfold triton kernel.

### CODE REVIEW

File: apex/contrib/test/openfold_triton/test_openfold_mha.py [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+    assert torch.allclose(ref_out, tri_out, atol=1e-2, rtol=0)
+    assert torch.allclose(ref_dv, tri_dv, atol=1e-2, rtol=0)
+    assert torch.allclose(ref_dk, tri_dk, atol=1e-2, rtol=0)
+    assert torch.allclose(ref_dq, tri_dq, atol=1e-2, rtol=0)
+    assert torch.allclose(ref_dbias, tri_dbias, atol=1e-2, rtol=0)
+
+
+if __name__ == "__main__":
+    # representative workload in openfold
+    test_openfold_triton_mha(Z=256, H=4, N_CTX=256, D_HEAD=32, dtype=torch.bfloat16)
```
Comment (crcrpar) [inline apex/contrib/test/openfold_triton/test_openfold_mha.py:111]: To clarifying my understanding, are the target devices A100 and H100?

**Review by crcrpar — COMMENTED** [review]
Yet to completely digest the content but it looks good overall

File: apex/contrib/test/openfold_triton/test_openfold_mha.py (hunk 1 above)
Comment (azazhu) [reply apex/contrib/test/openfold_triton/test_openfold_mha.py:111]: yes, The kernel can work on other chips. A100 and H100 have been well-tune.

File: apex/contrib/test/openfold_triton/test_openfold_mha.py [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+    tri_dk, k.grad = k.grad.clone(), None
+    tri_dq, q.grad = q.grad.clone(), None
+    tri_dbias, bias.grad = bias.grad.clone(), None
+
+    # check results
+    assert torch.allclose(ref_out, tri_out, atol=1e-2, rtol=0)
+    assert torch.allclose(ref_dv, tri_dv, atol=1e-2, rtol=0)
+    assert torch.allclose(ref_dk, tri_dk, atol=1e-2, rtol=0)
+    assert torch.allclose(ref_dq, tri_dq, atol=1e-2, rtol=0)
+    assert torch.allclose(ref_dbias, tri_dbias, atol=1e-2, rtol=0)
```
[the remark this answers is not in the document]
Comment (azazhu) [reply apex/contrib/test/openfold_triton/test_openfold_mha.py]: done

**Review by crcrpar — APPROVED** [review]

**Pull request merged**